FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -67,8 +69,10 @@ namespace edm {
   class WillGetIfMatch;
 
   namespace eventsetup {
+    struct ComponentDescription;
     class ESRecordsToProductResolverIndices;
-  }
+    class EventSetupProvider;
```

Comment:
I do not like everything suddenly becoming dependent upon EventSetupProvider. What appears to be needed is just a way to lookup for each EventSetup Record index what ComponentDescriptions are associated with a given index. I'd prefer a dedicated structure to lookup such information (could be a virtual interface which just goes back to the EventSetupProvider hidden inside of it).

In fact, maybe just returning a `std::map<ESRecordIndex, std::vector<ESProductResolverIndex>>` and then have the call which setups the data for the Service callback use that might be sufficient.

Reply:
Dependence removed. See comment above.